Process UV rotation even in full transform

unknownbrackets opened the pull request:
Assuming we don't expect these to be common, probably adding bits and etc. will just be a loss.  Or at least that's my excuse for just transforming inside the RotateUV func.

Fixes #3902.  I checked a few games which had broken before, and they work.  I don't think I have any games that use this feature, though.

-[Unknown]

unknownbrackets committed:
Process UV rotation even in full transform.

hrydgard commented on GPU/Common/SoftwareTransformCommon.cpp:
```diff
@@ -58,18 +58,24 @@ static void SwapUVs(TransformedVertex &a, TransformedVertex &b) {
+	Vec4f br;
+	Vec3ByMatrix44(br.AsArray(), v[0].pos, flippedMatrix);
+
+	float x1 = tl.x / tl.w;
```
you could save two divisions by getting the reciprocals of tl.w and br.w and multiplying instead, although the perf improvement would probably be negligible..

unknownbrackets replied:
Okay, changed it.  Also added a not working software renderer change, but it wasn't working before either.  Hmm..

-[Unknown]

unknownbrackets committed:
software: Examine v0/v1 for uv rotation.

If we determine tl/br heuristically, it can never rotate correctly.

Anyway, this still is not rotating correctly...

unknownbrackets committed:
Avoid some division.

lp200 commented:
![](http://i.imgur.com/Tiw4NUl.jpg)

:smiley:

thedax commented:
Always nice to see games getting fixes.
